lesson_06/task_1/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+func requestInfo(w http.ResponseWriter, req *http.Request) {
+	var ins response
+
+	ins.Host = req.Host
+	v := req.Header["Accept"]
+	v1 := req.Header["User-Agent"]
+	ins.Headers.UserAgent = v1
+	ins.Headers.Accept = v
+	ins.RequestURI = req.RequestURI
+	ins.UserAgent = v1[0]
+
+	js, err := json.Marshal(ins)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(js)
+}
+
+type requestHeaders struct {
+	Accept    []string `json:"Assept"`
+	UserAgent []string `json:"User-Agent"`
+}
+
+type response struct {
+	Host       string `json:"host"`
+	UserAgent  string `json:"user_agent"`
+	RequestURI string `json:"request_uri"`
+	Headers    requestHeaders
+}
+
+func main() {
+	http.HandleFunc("/headers", requestInfo)
+
+	http.ListenAndServe(":8081", nil)
```

Comment:
You should take care of errors, that happens here. Because if port if unavailable, the program will just ends, this is not normal. See how I do that staff here https://github.com/nikonok/go-course